test/org/zaproxy/zap/extension/pscanrules/TestInfoPrivateAddressDisclosureUnitTest.java

```diff
@@ -0,0 +1,334 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Copyright 2016 The ZAP Development Team
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.pscanrules;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertThat;
+
+import org.junit.Test;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.pscan.PluginPassiveScanner;
+
+public class TestInfoPrivateAddressDisclosureUnitTest extends PassiveScannerTest {
+	private static final String URI = "https://www.example.com/";
+
+	@Override
+	protected PluginPassiveScanner createScanner() {
+		return new TestInfoPrivateAddressDisclosure();
+	}
+
+	@Test
+	public void alertsIfPrivateIp() throws HttpMalformedHeaderException {
+		// ip as candidate / evidence
+		String[][] data = new String[][] {
+				// IPs defined in RFC 1918
+				{ "10.0.0.0",           "10.0.0.0" },
+				{ "10.10.10.10",        "10.10.10.10" },
+				{ "10.255.255.255",     "10.255.255.255" },
+
+				{ "172.16.0.0",         "172.16.0.0" },
+				{ "172.25.16.32",       "172.25.16.32" },
+				{ "172.31.255.255",     "172.31.255.255" },
+
+				{ "192.168.0.0",        "192.168.0.0" },
+				{ "192.168.36.127",     "192.168.36.127" },
+				{ "192.168.255.255",    "192.168.255.255" },
+
+				// some other stuff
+				{ "10.0.0.0:",          "10.0.0.0" },
+				{ "10.0.0.0:6553",      "10.0.0.0:6553" },
+				{ " 10.0.0.0 ",         "10.0.0.0" },
+				{ "/10.0.0.0.",         "10.0.0.0" },
+				{ ";10.0.0.0,",         "10.0.0.0" },
+				{ "\n10.0.0.0\t",       "10.0.0.0" },
+				{ "\n10.0.0.0\t",       "10.0.0.0" },
+
+				{ "10.0.0.0:bla",       "10.0.0.0" },
+				{ "15.10.0.0.0.12.27",  "10.0.0.0" },
+				{ "100.10.0.0.0.10.12", "10.0.0.0" }, 
+				{ "255.10.0.0.0:6555",  "10.0.0.0:6555" },
+
+				{ "10.0.0.0:128",       "10.0.0.0:128" },
+				{ "ip-10.0.0.0",        "10.0.0.0" },
+				{ "IP-10.0.0.0",        "10.0.0.0" }
+		};
+		for (int i = 0; i < data.length; i++) {
+			String candidate = data[i][0];
+			String evidence = data[i][1];
+			HttpMessage msg = createHttpMessage(candidate);
+			rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+			assertThat(alertsRaised.size(), equalTo(i + 1));
```

Comment:
It would be good to include the `candidate` as reason when asserting the sizes, it's easier to know which IP is failing (if that happens someday).